test/test_generator.rb

```diff
@@ -0,0 +1,47 @@
+require "find"
+require "yaml"
+
+require "minitest/autorun"
+
+require 'util/generator_helpers'
+
+class TestGenerator < Minitest::Test
+  include GeneratorTestHelpers
+
+  def setup
+    @default_options = [
+      'centos6-64mdca',
+    ]
+  end
+
+  def test_default_options
+    hash = run_cli_with_options(@default_options)
+    hosts = hash['HOSTS']
+
+    assert_includes(hosts, "centos6-64-1")
+    centos6 = hosts['centos6-64-1']
+    ["master", "database", "dashboard", "agent"].each do |role|
+      assert_includes(centos6['roles'], role)
+    end
+  end
+
+  def test_pe_dir_option
+    skip "pe_dir option not yet tested"
+  end
+
+  def test_fixtures
+    Find.find 'test/fixtures' do |f|
+      if File.directory?(f)
+        next
+      end
+
+      fixture_file = File.open(f, "r")
+      fixture_hash = YAML.load(fixture_file)
```

Comment:
Idiomatically, I would probably drop the temp file variable and go with:

``` ruby
fixture_hash = YAML.load_file(f)
```

or

``` ruby
fixture_hash = YAML.load(File.read(f))
```

(The "or" because I can't remember if `load_file` is always available, may depend on YAML version or implementation).